src/main/java/uk/gov/hmcts/ccd/domain/service/message/CaseEventMessageService.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.ccd.domain.service.message;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.casedetails.CaseAuditEventRepository;
+import uk.gov.hmcts.ccd.data.message.MessageCandidateRepository;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.aggregated.IdamUser;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseEventDefinition;
+import uk.gov.hmcts.ccd.domain.model.std.AuditEvent;
+import uk.gov.hmcts.ccd.domain.model.std.MessageInformation;
+import uk.gov.hmcts.ccd.domain.model.std.MessageQueueCandidate;
+
+import javax.inject.Inject;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+@Service
+@Qualifier("default")
+public class CaseEventMessageService implements MessageService {
```

Comment:
considering in the future we might have more classes like this one, we could consider something like this also:
`public class CaseEventMessageService extends AbstractMessageService`
`public abstract class AbstractMessageService implements MessageService {`
so that we have a place (AbstractMessageService) where to put common code among all subclasses

Reply:
done 👍